src/ctia/stores/es/sighting.clj

```diff
@@ -1,28 +1,74 @@
 (ns ctia.stores.es.sighting
-  (:require
-   [schema.core :as s]
-   [ctia.stores.es.crud :as crud]
-   [ctia.stores.es.query :refer [sightings-by-observables-query]]
-   [ctim.schemas.sighting :refer [Sighting
-                                  NewSighting
-                                  StoredSighting]]
-   [ctim.schemas.indicator :refer [Indicator]]
-   [ctia.lib.es.document :refer [search-docs]]))
-
-
-(def handle-create-sighting (crud/handle-create :sighting StoredSighting))
-(def handle-read-sighting (crud/handle-read :sighting StoredSighting))
-(def handle-update-sighting (crud/handle-update :sighting StoredSighting))
+  (:require [ctia.lib.pagination :refer [list-response-schema]]
+            [ctia.stores.es.crud :as crud]
+            [ctim.schemas
+             [common :refer [Observable]]
+             [sighting :refer [StoredSighting]]]
+            [schema-tools.core :as st]
+            [schema.core :as s]))
+
+(s/defschema ESStoredSighting
+  (st/merge StoredSighting
+            {:observables_hash [s/Str]}))
+
+(def es-coerce! (crud/coerce-to-fn [(s/maybe ESStoredSighting)]))
+
+(s/defn observable->observable-hash :- s/Str
+  [{:keys [type value] :as obsrvable :- Observable}]
+  (str type ":" value))
+
+(s/defn stored-sighting->es-stored-sighting :- (s/maybe ESStoredSighting)
+  "adds an observables hash to a sighting"
+  [{:keys [observables] :as s :- (s/maybe StoredSighting)}]
+  (when s
+    (assoc s :observables_hash
+           (map #(observable->observable-hash %) observables))))
+
+(s/defn es-stored-sighting->stored-sighting :- (s/maybe StoredSighting)
+  "remove the computed observables hash from a sighting"
+  [s :- (s/maybe ESStoredSighting)]
+  (when s (dissoc s :observables_hash)))
+
+(s/defn handle-create-sighting :- StoredSighting
+  [state realized]
+  (let [create-fn (crud/handle-create :sighting ESStoredSighting)
+        transformed (stored-sighting->es-stored-sighting realized)]
+    (-> (create-fn state transformed)
+        es-stored-sighting->stored-sighting)))
+
+(s/defn handle-read-sighting :- (s/maybe StoredSighting)
+  [state id]
+  (let [read-fn (crud/handle-read :sighting ESStoredSighting)]
+    (-> (read-fn state id)
+        es-stored-sighting->stored-sighting)))
+
+(s/defn handle-update-sighting :- StoredSighting
+  [state id realized]
+  (let [update-fn (crud/handle-update :sighting ESStoredSighting)
+        transformed (stored-sighting->es-stored-sighting realized)]
+    (-> (update-fn state id transformed)
+        es-stored-sighting->stored-sighting)))
```

Comment:
To my eye I would prefer, also, it might save a bit of work not to generate the `update-fn` for each call. The same apply to creation:

~~~
(def update-fn (crud/handle-update :sighting ESStoredSighting)

(s/defn handle-update-sighting :- StoredSighting
  [state id realised]
  (->> realized
       stored-sighting->es-stored-sighting
       (update-fn state id)
       es-stored-sighting->stored-sighting))
~~~